chia/full_node/mempool_check_conditions.py

```diff
@@ -132,14 +134,9 @@ def mempool_assert_my_amount(condition: ConditionWithArgs, unspent: CoinRecord)
     return None
 
 
-def get_name_puzzle_conditions(
-    block_program: SerializedProgram, safe_mode: bool
-) -> Tuple[Optional[str], Optional[List[NPC]], Optional[uint64]]:
-    # TODO: allow generator mod to take something (future)
-    # TODO: write more tests
-    block_program_args = SerializedProgram.from_bytes(b"\x80")
-
+def get_name_puzzle_conditions(generator: BlockGenerator, safe_mode: bool) -> NPCResult:
     try:
+        block_program, block_program_args = setup_generator_args(generator)
```

Comment:
In line 147 below (it wouldn't let me comment there) I think we want `result.first()` so that `result` can be expanded with expansion data later in a soft fork, data that has nothing to do with any of these coin spends. (This would also change what generators are expected to return... just throw a `(c XXX 0)` around the old program `XXX`.)

Reply:
Thanks, let me take a look.

Reply:
Which file and function?

Reply:
I see what you are saying - I'll start implementing that tomorrow, and poke you if I need info.